### ISSUE TITLE
Console History

### ISSUE THREAD

**Issue by VasVadum** [Pathoschild/SMAPI#790]
Basically, the SMAPI console needs to have history to scroll through. If I leave the window at its default tiny size, I have virtually no history at all. I can't scroll up to see what errors it told me about when I loaded the game. In this image below, you'll see a scroll bar, but only because I had the window maximized before it loaded all the data. If I hadn't, there'd be no scroll bar.

![image](https://user-images.githubusercontent.com/1490797/125180760-e4e8ba00-e1cb-11eb-9ff7-6794a30061f6.png)

Basically, just give this thing the ability to keep history. This is on Windows 10. I had a case once where a mod was causing loading issues and I couldn't get the error because it'd disappear behind newer text too fast, and I found I could get it to keep more of the history if I extended the window height, or clicked maximize faster.

**Comment by Pathoschild** [comment]
Hi! That's a limitation of Windows itself. You can [configure the scrollback limit](https://superuser.com/q/378313), or check [the SMAPI log file](https://smapi.io/log/) to see the full output (including details not shown in the console by default).

**Comment by VasVadum** [comment]
Scrollback limit can't be increased, your modding tools here use a custom CMD window.

![image](https://user-images.githubusercontent.com/1490797/125441558-64b0f92d-b3c0-401d-a56f-4311005abaff.png)

![image](https://user-images.githubusercontent.com/1490797/125441691-a53e19d2-5e74-4242-bc5d-9766e5aa80e9.png)

I set buffer size to 5x just now, in CMD. 250 lines instead of 50.

Yours, the buffer size depends on the window size. If the window defaults at that small size and I forget to maximize it, it deletes anything that falls off screen, no matter what buffer size there is. I checked the buffer size when yours loads up, it was back at 50 again while CMD in Windows, stayed at 250 as I had set it. I don't know what the difference is between the two, but I use CMD as a quick way to look at the issues, because of the colored text, it can be very helpful to me to see what I need to fix and helps me find things better.

**Comment by Pathoschild** [comment]
@VasVadum The command buffer is a different thing, it doesn't affect scrollback. Try this:

1. Launch SMAPI like usual.
2. Click the top-left icon and choose _defaults_ (not _properties_).
3. On the _layout_ tab, change the Screen Buffer Size to a larger height (I have it set to 9001).
4. Exit and relaunch SMAPI and it'll be increased permanently.

![image](https://user-images.githubusercontent.com/230581/126426608-b24b5e00-b8ea-484f-837b-2f3d888166c1.png)

**Comment by VasVadum** [comment]
Ah, ok, that worked. Annoying Microsoft defaulted it to 30 and made the window small and for some reason your tool opened CMD up differently than any other time I've ever used CMD.
